cheatsheets/Nodejs_security_cheat_sheet.md

```diff
@@ -0,0 +1,534 @@
+});
+```
+
+### Listen to errors when using EventEmitter
+
+When using EventEmitter, errors can occur anywhere in the event chain. Normally, if an error occurs in an EventEmitter object, an error event which has an Error object as an argument is called. However, if there are no attached listeners to that error event, the Error object that is sent as an argument is thrown and becomes an uncaught exception. In short, if you do not handle errors within an EventEmitter object properly, these unhandled errors may crash your application. Therefore, you should always listen to error events when using EventEmitter objects.
+
+```JavaScript
+var events = require('events');
+var myEventEmitter = function(){
+    events.EventEmitter.call(this);
+}
+require('util').inherits(myEventEmitter, events.EventEmitter);
+myEventEmitter.prototype.someFunction = function(param1, param2) {
+    //in case of an error
+    this.emit('error', err);
+}
+var emitter = new myEventEmitter();
+emitter.on('error', function(err){
+    //Perform necessary error handling here
+});
+```
+
+### Handle errors in asynchronous calls
+
+Errors that occur within asynchronous callbacks are easy to miss. Therefore, as a general principle first argument to the asynchronous calls should be an Error object. Also, express routes handle errors itself, but it should be always remembered that errors occurred in asynchronous calls made within express routes are not handled, unless an Error object is sent as a first argument.
+
+Errors in these callbacks can be propagated as many times as possible. Each callback that the error has been propagated to can ignore, handle or propagate the error.
+
+## Server Security
+
+### Set cookie flags appropriately
+
+Generally, session information is sent using cookies in web applications. However, improper use of HTTP cookies can render an application to several session management vulnerabilities. There are some flags that can be set for each cookie to prevent these kinds of attacks. `httpOnly`, `Secure` and `SameSite` flags are very important for session cookies. `httpOnly` flag prevents the cookie from being accessed by client-side JavaScript. This is an effective counter-measure for XSS attacks. `Secure` flag lets the cookie to be sent only if the communication is over HTTPS. `SameSite` flag can prevent cookies from being sent in cross-site requests which helps protect against Cross-Site Request Forgery (CSRF) attacks. Apart from these, there are other flags like domain, path and expires. Setting these flags appropriately is encouraged, but they are mostly related to cookie scope not the cookie security. Sample usage of these flags is given in the following example:
+
+```JavaScript
+var session = require('express-session');
+app.use(session({
+    secret: 'your-secret-key',
+    key: 'cookieName',
+    cookie: { secure: true, httpOnly: true, path: '/user', sameSite: true}
+}));
+```
+
+### Use appropriate security headers
```

Comment:
I'd recommending removing lots of the wordings around each header, and linking to the MDN official documentation.